common/src/main/java/xyz/wagyourtail/jsmacros/client/api/library/impl/FUtil.java

```diff
@@ -0,0 +1,496 @@
+import xyz.wagyourtail.jsmacros.client.api.helpers.BossBarHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.ChatHudLineHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.CheckBoxWidgetHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.ChunkHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.ClickableWidgetHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.CommandContextHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.CyclingButtonWidgetHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.DirectionHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.EnchantmentHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.EntityHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.FluidStateHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.FoodComponentHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.FormattingHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.FullOptionsHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.ItemStackHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.LockButtonWidgetHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.NBTElementHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.PacketByteBufferHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.PlayerAbilitiesHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.PlayerListEntryHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.RecipeHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.ScoreboardObjectiveHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.ScoreboardsHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.ServerInfoHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.SliderWidgetHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.StatsHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.StatusEffectHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.StyleHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.SuggestionsBuilderHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.TeamHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.TextFieldWidgetHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.TextHelper;
+import xyz.wagyourtail.jsmacros.client.util.NameUtil;
+import xyz.wagyourtail.jsmacros.core.helpers.BaseHelper;
+import xyz.wagyourtail.jsmacros.core.library.BaseLibrary;
+import xyz.wagyourtail.jsmacros.core.library.Library;
+import xyz.wagyourtail.wagyourgui.elements.CheckBox;
+import xyz.wagyourtail.wagyourgui.elements.Slider;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.SplittableRandom;
+
+/**
+ * @author Etheradon
+ * @since 1.8.4
+ */
+@Library("Util")
```

Comment:
maybe call this one `JavaUtils` idk

Reply:
I'm not really sure what an appropiate name for this library would be as it's purpose is to add miscellaneous methods that don't belong to any other Library.

Reply:
ok. on closer look. yeah this really seems like a combination of doing stuff with raw java, and some general util stuff.
so splitting to `JavaUtils` and `Utils` maybe. or some of the non raw java stuff could be in different libraries idk

Reply:
There are some other smaller libraries like Time, Request, KeyBind or PositionCommon, so I don't think that should be a problem. Splitting the library definitely makes sense.
I marked the open methods in the JsMacros library as deprecated and linked to the new util library, as these methods don't belong to JsMacros itself. It may also be a good idea to move the playSound methods from World, because 4 of them don't even have anything to do with the world and it doesn't seem quite right to leave them in there.